### ISSUE TITLE
Add support for drag and drop of text selections

### ISSUE THREAD

**Issue by eyelessjerry** [glushchenko/fsnotes#1591]
macOS Ventura Apple silicon

I cannot drag and drop any text to notes – is it by design? Guess that would be like the second thing one would try in using the program ... 🤔. (At first I had problem copying and pasting text to fsnotes also, but upon restart (moved the directory to iCloud) it works.)

**Comment by eyelessjerry** [comment]
And, for god's sake – windows should minimize when double-clicked by default ... at least there should be an option to use the default behaviour (it used to be the default behaviour in System Settings, but intelligent people still set it as their default in System settings and expect it to be honored in all apps, unless they opt out of it ...).

**Comment by eyelessjerry** [comment]
I also cannot figure out the difference between using the .md & .markdown formats – they are supposed to be the same, but maybe not here ... .

**Comment by gingerbeardman** [comment]
lots to address in these three

1. drag and drop: not quite sure what you're trying. you can drag and drop files onto the app. for text selections: copy and paste can be done.

2. i will file this as another issue

3. right, .md and .markdown, no difference. some people prefer the shorter extension some prefer the longer.

**Comment by eyelessjerry** [comment]
Yes, I have always been dragging text between windows in all apps I use – I know it does not necessarily always works, but it is often easier than copying and pasting (even when knowing the shortcuts by heart). It is not so bloody important, but I find it disturbing when one cannot drag text – which one also use inside document. (Just the other day filed a bug with Apple who doesn't allow it in the Messages app either, but that doesn't make their way of abandoning making macOS easy to use an argument for others to do the same.) (I specifically said "text", not files, which I tested also.)

**Comment by shimeike** [comment]
Is this functionality (dragging selected text within the fsnotes editor) a possibility on macos version? My wife has just had to switch from nvalt and misses this functionality (which appears to work in many/most text fields). Great app - thanks!

**Comment by eyelessjerry** [comment]
Generally it is a possibility on macOS and in apps. I can select words in the macOS Notes app and drag to FSNotes and nothing happen, but dragging the other was from FSNotes to Notes works as expected.

/Jerry

**Comment by gingerbeardman** [comment]
It seems FSNotes needs to advertise or activate its ability to receive dragged text.

**Comment by shimeike** [comment]
To clarify, the use case here is dragging and dropping text within the FSNotes editor window itself (not necessarily from another application). This seems to work in TextEdit  on latest OS, and it also worked in NVAlt on an old Intel machine previously.

The reason for wanting this (I'm told) is to achieve the same as cut/paste, but without putting the moved text into the clipboard manager ...

**Comment by gingerbeardman** [comment]
Sure, understood. I can do that, so what are you seeing? @shimeike 


https://github.com/user-attachments/assets/81276e53-f545-4cf2-94c0-cc82445f0eec



(also, do you know about Clipboard Managers that go over and above the system one? I use the one in Alfred, and I can set it to ignore apps so it would not manage anything I copy and paste in FSNotes - by default it ignres all common password apps for obvious security reasons)

**Comment by eyelessjerry** [comment]
That's funny ... I could not do that in any of the notes I had since long back in FSNotes, but then decided to double-check, creating a new one and I could now move text in the note ... hmm.

I started using Yoink recently as a clipboard manager and now saw I could edit which app to ignore there. https://eternalstorms.at

**Comment by gingerbeardman** [comment]
@eyelessjerry congrats! you just found a bug, I can repo: 

- https://github.com/glushchenko/fsnotes/issues/1771

**Comment by shimeike** [comment]
> But I can do that, so what are you seeing?

That is interesting, @gingerbeardman . I seem to have the same behaviour as @eyelessjerry . In a newly created note within FSNotes, it works per your screencast. If I select another note, then the dragging/dropping ceases to function. So the text view of a newly created note appears to behave differently.

Yes, we use jumpcut as a clipboard manager. The desire from my "user who is always right" (!) is to have the ability to sometimes move text around without taking a slot in the jumpcut history.

**Comment by gingerbeardman** [comment]
See my report for the new bug

- #1771

**Comment by gingerbeardman** [comment]
I've turned this into a bug in light of it still being an issue after #1771 has been addressed

**Comment by glushchenko** [comment]
[FSNotes.zip](https://github.com/user-attachments/files/17194855/FSNotes.zip)

@eyelessjerry please check test build.

**Comment by eyelessjerry** [comment]
Everything seems to work as expected in this version!

**Comment by shimeike** [comment]
Agreed - dragging and dropping text works as expected with this version. Thanks!
